File: backend/app/schemas/library.py
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel


class LibraryMediaItem(BaseModel):
    id: int
    source_type: str
    event_id: int
    event_name: str
    event_type: str | None
    event_date: date | None
    title: str
    filename: str
    media_type: str
    file_type: str | None
    status: str
    local_path: str
    relative_path: str | None
    thumbnail_url: str | None
    file_exists: bool
    width: int | None
    height: int | None
    duration_seconds: float | None
    original_media_id: int | None
    curated_media_id: int | None
    enhanced_media_id: int | None
    created_at: datetime


class LibraryMediaListResponse(BaseModel):
    items: list[LibraryMediaItem]


class LibraryPieceItem(BaseModel):
    id: int
    event_id: int
    event_name: str
    event_type: str | None
    event_date: date | None
    piece_type: str
    title: str
    purpose: str | None
    target_platform: str | None
    aspect_ratio: str | None
    status: str
    output_path: str | None
    absolute_output_path: str | None
    thumbnail_url: str | None
    media_count: int
    copy_count: int
    approved_copy_count: int
    created_at: datetime
    updated_at: datetime


class LibraryPieceListResponse(BaseModel):
    items: list[LibraryPieceItem]


class LibraryCopyItem(BaseModel):
    id: int
    piece_id: int
    event_id: int
    event_name: str
    event_type: str | None
    event_date: date | None
    piece_title: str
    copy_type: str
    variant_label: str | None
    body_preview: str
    status: str
    output_path: str | None
    absolute_output_path: str | None
    created_at: datetime
    updated_at: datetime


class LibraryCopyListResponse(BaseModel):
    items: list[LibraryCopyItem]


class LibrarySearchItem(BaseModel):
    entity_type: str
    id: int
    event_id: int
    event_name: str
    title: str
    subtitle: str | None
    status: str
    date: date | datetime | None
    local_path: str | None
    thumbnail_url: str | None


class LibrarySearchResponse(BaseModel):
    items: list[LibrarySearchItem]

File: desktop/src/types/library.ts
export type LibraryMediaItem = {
  id: number;
  source_type: string;
  event_id: number;
  event_name: string;
  event_type: string | null;
  event_date: string | null;
  title: string;
  filename: string;
  media_type: string;
  file_type: string | null;
  status: string;
  local_path: string;
  relative_path: string | null;
  thumbnail_url: string | null;
  file_exists: boolean;
  width: number | null;
  height: number | null;
  duration_seconds: number | null;
  original_media_id: number | null;
  curated_media_id: number | null;
  enhanced_media_id: number | null;
  created_at: string;
};

export type LibraryMediaListResponse = {
  items: LibraryMediaItem[];
};

export type LibraryPieceItem = {
  id: number;
  event_id: number;
  event_name: string;
  event_type: string | null;
  event_date: string | null;
  piece_type: string;
  title: string;
  purpose: string | null;
  target_platform: string | null;
  aspect_ratio: string | null;
  status: string;
  output_path: string | null;
  absolute_output_path: string | null;
  thumbnail_url: string | null;
  media_count: number;
  copy_count: number;
  approved_copy_count: number;
  created_at: string;
  updated_at: string;
};

export type LibraryPieceListResponse = {
  items: LibraryPieceItem[];
};

export type LibraryCopyItem = {
  id: number;
  piece_id: number;
  event_id: number;
  event_name: string;
  event_type: string | null;
  event_date: string | null;
  piece_title: string;
  copy_type: string;
  variant_label: string | null;
  body_preview: string;
  status: string;
  output_path: string | null;
  absolute_output_path: string | null;
  created_at: string;
  updated_at: string;
};

export type LibraryCopyListResponse = {
  items: LibraryCopyItem[];
};

export type LibrarySearchItem = {
  entity_type: string;
  id: number;
  event_id: number;
  event_name: string;
  title: string;
  subtitle: string | null;
  status: string;
  date: string | null;
  local_path: string | null;
  thumbnail_url: string | null;
};

export type LibrarySearchResponse = {
  items: LibrarySearchItem[];
};

export type LibraryQuery = {
  event_id?: number;
  date_from?: string;
  date_to?: string;
  event_type?: string;
  file_type?: string;
  piece_type?: string;
  copy_type?: string;
  status?: string;
  q?: string;
  source_type?: string;
  limit?: number;
};
